readium/shared/src/main/java/org/readium/r2/shared/util/Try.kt

```diff
@@ -113,6 +113,20 @@ public fun <S, F : Throwable> Try<S, F>.getOrThrow(): S =
         is Failure -> throw value
     }
 
+/**
+ * Returns the encapsulated value if this instance represents success
+ * or throws the encapsulated ThrowableError exception if it is failure.
+ */
+@Suppress("UnusedReceiverParameter")
+@Deprecated(
+    "An `Error` is not a throwable object. Refactor or wrap in an `ErrorException`.",
+    ReplaceWith("getOrElse { throw ErrorException(it) }"),
+    DeprecationLevel.ERROR
+)
+@JvmName("getOrThrowError")
+public fun <S, F : Error> Try<S, F>.getOrThrow(): S =
+    throw NotImplementedError()
+
```

Comment:
I doubt this could actually appear to API consumers because there were no such things as `Error`s before (if I remember correctly).

Reply:
We have a few APIs that use to return `Try<X, Exception>` that now return `Try<X, Error>`. This will suggest migrating to a proper handling of errors, or to rethrow a `ErrorException`.

Reply:
Got it.